src/app/parser/open-api/header-parser.ts

```diff
@@ -18,3 +22,25 @@ export function parseHeaders(
   }
   return headerMap;
 }
+
+export function parseRouteHeaders(
+  route: routeTypes.ApiRoute,
+): routeTypes.ParsedHeaders | undefined {
+  if (!route.headersObjectSchema) {
+    return undefined;
+  }
+  if (!schemaTypes.schemaPropertyIsTypeObject(route.headersObjectSchema)) {
+    const basicChars = routeTypes.getBasicCharacteristics(route);
+    logger.error(
+      `Cannot parse headers for API Route: ${basicChars.method} ${basicChars.route}\nERROR: route.headersObjectSchema is not of type object`,
+    );
+    return undefined;
+  }
+  const headers: Set<string> = new Set<string>(
+    Object.keys(route.headersObjectSchema.properties),
+  );
+  if (headers.size === 0) {
+    return undefined;
+  }
```

Comment:
Why are we doing this?

Reply:
The follow up PR to this, which will link the codegen with this header parser will have a check to only add headers to an api if the parser does not return `undefined`. That's why I wanted to make this condition explicit.